src/services/schedulingService.ts

```
@@ -51,393 +57,446 @@ export const getEffortPoints = (effortLevel: EffortLevel): number => {
+
+      const updatedTask = await taskService.updateTask(task.id, updates);
+      console.log(`[SchedulingService.scheduleTask] Updated large task ${task.id} in DB:`, updatedTask);
+      return updatedTask;
+    } else { 
+      console.warn(`[SchedulingService.scheduleTask] Could not schedule large task ${task.id}. No segments were created or task could not be scheduled. Returning null.`);
+      return null; 
     }
-    return taskToSchedule; // Already in a final state or no action needed
   }
+  // Fallback, though ideally all paths above should return.
+  return null;
```

Biome (1.9.4)

[error] This code is unreachable

... because either this statement, ...

... this statement, ...

... this statement, ...

... or this statement will return from the function beforehand


(lint/correctness/noUnreachable)